Make use of cpuid.h optional at configure time

cmcantalupo opened the pull request:
merge after #3296 because this PR removes the ExclusiveArch flag from the spec files.

cmcantalupo committed:
Abstract use of cpuid

- Remove unused includes of cpuid.h
- Add a configure option --disable-cpuid

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd commented on service/test/CpuinfoIOGroupTest.cpp:
```diff
@@ -32,6 +21,7 @@ extern "C"
 #include "geopm/PluginFactory.hpp"
 #include "geopm/PlatformTopo.hpp"
 #include "geopm_test.hpp"
+#include "MockCpuid.hpp"
```
I think this newly-added header is unused. Did you plan to add the new cpuid class to this the CpuinfoIOGroup interface, or is the header left over from an earlier design?

cmcantalupo replied:
Left overs! Thanks

cmcantalupo committed:
Fix memory error using free directly in IOUringImp

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Add support for clusters in place of sockets in arm lscpu output

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dannosliwcd commented on service/configure.ac:
```diff
@@ -454,6 +482,10 @@ if test "x$enable_io_uring" = "x1" ; then
          #include <liburing.h>])
+    AC_CHECK_LIB([uring], [io_uring_free_probe], [liburing_has_free=1], [liburing_has_free=0])
+    if test "x$liburing_has_free" = "x1"; then
+        AC_DEFINE([GEOPM_IO_URING_HAS_FREE], [ ], [Use io_uring_free_probe instead of free])
```
Maybe add a comment about this since there's no issue to track the problem we encountered. I guess key bits are that some versions of liburing expect us to use free and other versions give us the free wrapper. When we use the wrong one, the program aborts.

dannosliwcd commented on service/test/CpuinfoIOGroupTest.cpp:
```diff
@@ -10,17 +10,6 @@
-#define __get_cpuid(p0, p1, p2, p3, p4) mock_cpuid(p0, p1, p2, p3, p4)
-}
 #define GEOPM_TEST
 #include "src/CpuinfoIOGroup.cpp"
```
I suspect we don't need this "include the implementation" pattern here any more. This is pre-existing, so I don't think it must be addressed now (maybe record in a tech debt issue). It might still be needed if any of the test cases rely on parts of the interface hidden from the header.

I think the reason we needed it before was so we could map the cpuid functions to our mock implementations at preprocessing time. Now that we go through an object wrapper, we can use our usual execution-time dependency injection instead. It might still be needed if any of the test cases rely on parts of the interface hidden from the header.

This is pretty low priority. I think the main related tech-debt issue is that these kinds of implementation dependencies would make it more difficult for us to separate public and non-public components of our interface.

cmcantalupo committed:
Remove include of source code file in test

- Add comment to configure logic describing why
  GEOPM_IO_URING_HAS_FREE is needed.

Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>
